pandas/io/s3.py

```diff
@@ -25,7 +26,15 @@ def get_file_and_filesystem(
     if mode is None:
         mode = "rb"
 
-    fs = s3fs.S3FileSystem(anon=False)
+    # Support customised S3 servers endpoint URL via environment variable
+    # The S3_ENDPOINT should be the complete URL to S3 service following
+    # the format: http(s)://{host}:{port}
+    s3_endpoint = os.environ.get('S3_ENDPOINT')
```

Comment:
Part of the CI process checks that the code would be left unchanged by the [black autoformatter](https://github.com/psf/black).
Try running
```
black pandas/io/s3.py --diff
```
to see the changes (and leave out the `--diff` flag to have them applied)